pkg/arvo/gen/hood/moon.hoon

```diff
@@ -0,0 +1,44 @@
+::  Create a private key-file for a random (or specified) moon
+::
+::::  /hoon/moon/hood/gen
+  ::
+/-  *sole
+/+  *generators
+::
+::::
+  ::
+:-  %say
+|=  $:  [now=@da eny=@uvJ bec=beak]
+        arg=?(~ [mon=@p ~])
+        public-key=pass
+    ==
+:-  %helm-moon
+^-  (unit [=ship =udiff:point:able:jael])
+=*  our  p.bec
+=/  ran  (clan:title our)
+?:  ?=({?($earl $pawn)} ran)
+  %-  %-  slog  :_  ~
+      leaf+"can't create a moon from a {?:(?=($earl ran) "moon" "comet")}"
+  ~
+=/  mon=ship
+  ?^  arg
+    mon.arg
+  (add our (lsh 5 1 (end 5 1 (shaz eny))))
+=/  seg=ship  (sein:title our now mon)
+?.  =(our seg)
+  %-  %-  slog  :_  ~
+      :-  %leaf
+      "can't create keys for {(scow %p mon)}, which belongs to {(scow %p seg)}"
+  ~
+=/  =pass
+  ?.  =(*pass public-key)
+    public-key
+  =/  cub  (pit:nu:crub:crypto 128 (sham mon life=1 eny))
```

Comment:
I'm slightly partial to using 512 bits of entropy ("more is better"). And it seems a little strange to to take a 128-bit hash, and then take a 512-bit hash of it (in `+pit:nu:crub`). But I'm also not sure it's an actual problem. Is this just to align 128-bit addresses with 128-bit seeds?

Reply:
nah, it's just the only hash function that takes a noun instead of an atom.  I've replaced it with `(shaz (jam mon life eny))`, which should give us 512 bits.